source/MaterialXGenShader/Util.cpp

```diff
@@ -636,14 +636,19 @@ bool elementRequiresShading(ConstTypedElementPtr element)
 
 vector<NodePtr> getShaderNodes(const NodePtr materialNode, const string& shaderType, const string& target)
 {
-    DocumentPtr doc = materialNode->getDocument();
+    ElementPtr parent = materialNode ? materialNode->getParent() : nullptr;
```

Comment:
This would fail if the materialnode was inside a nodegraph since the attached shader nodes would never be found. Note that if hierarchical nodegraphs are supported this code still works since there is no restriction on where a possible upstream nodegraph can exist other than the fact it is a sibling of the materialnode.